Tests/LegacyUnit/Service/RegistrationManagerTest.php

```diff
@@ -321,6 +318,45 @@ private function filterEmailAttachmentsByTitle(MailMessage $email, string $conte
         return $matches;
     }
 
+    /**
+     * Adds an instance to the Typo3 instance FIFO buffer used by `GeneralUtility::makeInstance()`
+     * and registers it for purging in `tearDown()`.
+     *
+     * In case of a failing test or an exception in the test before the instance is taken
+     * from the FIFO buffer, the instance would stay in the buffer and make following tests
+     * fail. This function adds it to the list of instances to purge in `tearDown()` in addition
+     * to `GeneralUtility::addInstance()`.
+     *
+     * @param string $className
+     * @param mixed $instance
+     *
+     * @return void
+     */
+    private function addMockedInstance(string $className, $instance)
+    {
+        GeneralUtility::addInstance($className, $instance);
+        $this->mockedClassNames[] = $className;
+    }
+
+    /**
+     * Purges possibly leftover instances from the Typo3 instance FIFO buffer used by
+     * `GeneralUtility::makeInstance()`.
+     *
+     * @return void
+     */
+    private function purgeMockedInstances()
+    {
+        if (empty($this->mockedClassNames)) {
+            return;
+        }
```

Comment:
We don't really need this guard clause here as a `foreach` on an empty array is a no-op. (This will simplify the code a bit.)